src/libstd/ffi/c_str.rs

```diff
@@ -1551,6 +1551,17 @@ impl ops::Index<ops::RangeFull> for CString {
     }
 }
 
+#[stable(feature = "cstr_range_from", since = "1.45.0")]
+impl ops::Index<ops::RangeFrom<usize>> for CStr {
+    type Output = CStr;
+
+    #[inline]
+    fn index(&self, index: ops::RangeFrom<usize>) -> &CStr {
+        let bytes = self.to_bytes_with_nul();
+        unsafe { CStr::from_bytes_with_nul_unchecked(&bytes[index.start..]) }
```

Comment:
Seems like this would be UB if &amp;bytes\[index.start..\] is empty.

Reply:
Oh huh, didn't know ranges with a start equal to the length of the slice were valid.

I'll see what I can do.

Reply:
okay, this should be taken care of. basically used `self.to_bytes()` to intentionally exclude the null byte, so if the starting index points to where the null byte would be, then it gives the appropriate out-of-bounds panic message. if it's in-bounds, then i use `slice::from_raw_parts()` and `CStr::from_bytes_with_nul_unchecked()` to avoid recalculating the string length when we already know it's a valid `CStr`.